lib/OpenQA/FakeApp.pm

```diff
@@ -43,24 +47,53 @@ has schema => sub { OpenQA::Schema::connect_db() };
 
 sub setup_log {
     my ($self) = @_;
-    if ($self->log_dir) {
+    my ($logfile, $logdir, $level, $log);
+
+    if ($self->isa('OpenQA::FakeApp')) {
+        $logdir = $self->log_dir;
+        $level  = $self->level;
+    }
+    else {
+        $log = $self->log;
+        $level = $self->config->{logging}->{level} || 'debug';
+    }
+    $logfile = $ENV{OPENQA_LOGFILE} || $self->config->{logging}->{file};
+
+    if ($logfile && $logdir) {
+        my $logfile = catfile($logdir, $logfile);
+        $log = Mojo::Log->new(handle => \*STDOUT, level => $self->level);
+    }
+    elsif ($logfile) {
+        $log = Mojo::Log->new(handle => path($logfile)->open('>>'), level => $level);
+    }
+    elsif ($logdir) {
         # So each worker from each host get it's own log (as the folder can be shared). Hopefully the machine hostname
         # is already sanitized. Otherwise we need to check
-        my $logfile
+        $logfile
           = catfile($self->log_dir, hostname() . (defined $self->instance ? "-${\$self->instance}" : '') . ".log");
-
-        $self->log->handle(path($logfile)->open('>>'));
-
-        $self->log->path($logfile);
+        $log = Mojo::Log->new(handle => path($logfile)->open('>>'), level => $self->level);
     }
-
-    $self->log->format(
+    else {
+        $log = Mojo::Log->new(handle => \*STDOUT, level => $level);
+    }
+    $log->format(
```

Comment:
When logging to STDOUT/STDERR, output format should not contain time, see the portion you removed: https://github.com/os-autoinst/openQA/pull/1480/commits/1b9b39c2c48079bc31b67d6e63733dd1dd1b30c1#diff-5289bf936364851cb95d4833773bce49L132 - Also see https://progress.opensuse.org/issues/13866